ckanext/xloader/loader.py

```diff
@@ -79,6 +79,49 @@ def detect_encoding(file_path):
     return detector.result  # e.g. {'encoding': 'EUC-JP', 'confidence': 0.99}
 
 
+def _fields_match(fields, existing_fields, logger):
+    ''' Check whether all columns have the same names and types as previously,
+    independent of ordering.
+    '''
+    # drop the generated '_id' field
+    for index in range(len(existing_fields)):
+        if existing_fields[index]['id'] == '_id':
+            existing_fields.pop(index)
```

Comment:
removing an item from the list then indexing based on the old list length can cause an IndexError, right?

Reply:
We immediately `break` after popping it.

Reply:
yeah caught that just after posting my comment :-)